flink-baseline/flink-canary/src/main/java/com/ness/flink/canary/pipeline/function/HealthCheckFunction.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.kafka.common.Node;
+
+@Slf4j
+public class HealthCheckFunction extends ProcessFunction<KafkaConfigs, String>{
+    private static final long serialVersionUID = 1L;
+    @Override
+    public void processElement(KafkaConfigs value, ProcessFunction<KafkaConfigs, String>.Context ctx, Collector<String> out) throws Exception {
+        String result;
+
+        Properties props = new Properties();
+        String bootStrapServers = value.getBootStrapServers();
+        props.put("bootstrap.servers", bootStrapServers);
+        props.put("request.timeout.ms", value.getRequestTimeoutMs());
+        props.put("connections.max.idle.ms", value.getConnectionMaxIdleMs());
+
+        if (log.isInfoEnabled()) {
+            log.info("Broker : {} {}", bootStrapServers, props.get("bootstrap.servers"));
+        }
+
+        try (AdminClient adminClient = AdminClient.create(props)) {
+            if (verifyBrokerConnection(out, bootStrapServers, adminClient))  {
+                Set<String> topicNames = adminClient.listTopics().names().get();
+                boolean topicExists = topicNames.contains(value.getTopic());
+
+                if (topicExists) {
+                    result = "Kafka Topic: Exists - " + value.getTopic();
+
+                    if (log.isInfoEnabled()) {
+                        log.info(result);
+                    }
+                    out.collect(result);
+                }
+                else {
+                    result = "Kafka Topic: Not Found: - " + value.getTopic();
+
+                    if (log.isInfoEnabled()) {
+                        log.error(result);
+                    }
+                    out.collect(result);
+                }
+            }
+
+        }
+
+    }
+
+    private boolean verifyBrokerConnection(Collector<String> out, String bootStrapServers, AdminClient adminClient) {
+        String result;
+        boolean connectionSuccessful = false;
+        try {
+            connectionSuccessful = verifyConnection(adminClient);
+            if (connectionSuccessful) {
+                result = "Kafka Broker Connection: Successful - " + bootStrapServers;
+
+                if (log.isInfoEnabled()) {
+                    log.info(result);
+                }
+                out.collect(result);
+            }
+        } catch (ExecutionException | InterruptedException ex) {
+            result = "Kafka Broker Connection: Failed - " + bootStrapServers;
+
+            if (log.isInfoEnabled()) {
+                log.error(result);
+            }
+            out.collect(result);
+        }
+        return connectionSuccessful;
+    }
+
+    public boolean verifyConnection(AdminClient adminClient) throws ExecutionException, InterruptedException {
+        Collection<Node> nodes;
+        /*
+        control how long to try to connect to broker in ms here. default is 5000 ms or 5 seconds.
+         */
+        nodes = adminClient.describeCluster(new DescribeClusterOptions().timeoutMs(5000))
```

Comment:
Shouldn't be hardcoded.

Reply:
Done, kafkaAdminProperties now takes in a value for this so it is not hardcoded anymore.